app/frontend/wallet/account.ts

```diff
@@ -36,10 +36,18 @@ import blockchainExplorer from './blockchain-explorer'
 
 const DummyAddressManager = () => {
   return {
-    discoverAddresses: () => [],
-    discoverAddressesWithMeta: () => [],
-    getAddressToAbsPathMapping: () => ({}),
-    _deriveAddress: () => ({}),
+    discoverAddresses: (): Promise<_Address[]> => Promise.resolve([]),
+    discoverAddressesWithMeta: (): Promise<
+      {
+        address: _Address
+        bip32StringPath: string
+        isUsed: boolean
+      }[]
+    > => Promise.resolve([]),
+    getAddressToAbsPathMapping: (): {
+      [key: string]: BIP32Path
+    } => ({}),
```

Comment:
getAddressToAbsPathMapping returns this type
```
{
      [key: string]: BIP32Path
}
```
both here and in address-manager, I'd extract its type there and use it here as well, this is getting complicated.

Same goes for discoverAddressesWithMeta, that doesn't have any type in address-manager at all

Reply:
I didn't bother with this since this dummy address manager should be refactored somehow so I only typed the return statements because I needed them for the tx planning but ok, I ll type the functions. :+1: